CIP-????/README.md

```diff
@@ -0,0 +1,334 @@
+
+#### Atomic transaction groups
+
+The required transactions feature enables us to bind transactions together such that they must be included either all together or not at all. 
+We do this by creating a dependency loop, but without creating a loop at the level of value flow. 
+We can construct dependency loops either just using required transactions, or by using the dependencies created by normal value flow and augmenting them with required transactions (e.g. tx A spends an output of tx B; tx B requires tx A; now we have a loop and hence atomicity).
+
+The design is heavily inspired by [atomic transaction groups in Algorand](https://developer.algorand.org/docs/get-details/atomic_transfers/). 
+Algorand’s design is equivalent to ours with the additional restriction that every transaction in the “group” must require every other transaction. 
+We don’t want every transaction in a validity zone to require every other transaction, so if we were to use Algorand’s design we would need an additional kind of container for transactions to bound the atomic group.
+
+In contrast, by allowing transactions to have a specific list of stated dependencies, that means that it is legal for transactions to not state any dependencies, which is for us the normal case. 
+That means we can get away with only one kind of container: validation zones.
+
+This design also gives us some additional power, in that it lets us express dependency graphs other than connected components, but we do not currently know of a use for this.
+
+#### Handling validation failures
+
+The main problem with handling validation failures is that generally we need to either apply a zone in its entirety or not at all.
+
+That means that if we validate a transaction T1 early in a zone and then later hit a failure in a transaction T2, then we cannot just apply T1 (it might require T2!), which means we do not get the fees for the work done in validating T1.
+
+The first thing is to flush out cheap phase 1 failures. 
+We don’t want to discover a phase 1 failure late in the process that forces us to discard expensive transactions where we ran scripts. So we check all transactions for phase 1 failures first.
+
+Phase 2 failures are, as usual, annoying. 
+Of course, we want to claim the collateral from such a transaction if it fails. 
+But we need a way to take the collateral and prove that we did the right thing without applying any other transactions from the zone.
+
+Our first problem is how to get the collateral at all. 
+Rule Safe Collateral helps us here: by ensuring that collateral inputs come from outside the zone, we can be sure that we will actually be able to take them if we discard the zone.
+
+The second problem is how to ensure that the node is compensated for the work that it did validating transactions that came before the failing transaction in the zone. 
```

Comment:
I see the excess collateral rule more as "paying for space in the block that these witness transactions occupy", rather than having to do with paying for Phase-1 validation, compensation for which is not generally guaranteed

Reply:
Ah right, good point. If we were just compensating them for their work we would pay them much less. But what justifies the full fee is that we actually need to include those transactions in the chain.